monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -559,19 +559,31 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
+    @classmethod
+    def get_latest_event_timestamp(cls):
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
+
+        agent_events = cls._agent_event_repository.get_events()
+        latest_timestamp = max(agent_events, key=lambda event: event.timestamp).timestamp
+
+        return str(latest_timestamp)
+
     @staticmethod
     def is_latest_report_exists():
         """
         This function checks if a monkey report was already generated and if it's the latest one.
         :return: True if report is the latest one, False if there isn't a report or its not the
         latest.
         """
-        latest_report_doc = mongo.db.report.find_one({}, {"meta_info.latest_monkey_modifytime": 1})
+        latest_report_doc = mongo.db.report.find_one({}, {"meta_info.latest_event_timestamp": 1})
 
         if latest_report_doc:
-            report_latest_modifytime = latest_report_doc["meta_info"]["latest_monkey_modifytime"]
-            latest_monkey_modifytime = Monkey.get_latest_modifytime()
-            return report_latest_modifytime == latest_monkey_modifytime
+            report_latest_event_timestamp = latest_report_doc["meta_info"][
+                "latest_event_timestamp"
+            ].replace(",,,", ".")
```

Comment:
what is this replace doing?

Reply:
The timestamp looks like: 1231234.123442, so when we encode the query it replaces all dots with `,,,` so in order to compare them I need to decode it.

Reply:
The bigger question is, "why is the timestamp a string?"

Reply:
Well I tried converting the timestamp to `datetime` but our events differs in ~0.000002 seconds and datetime was not storing that much decimals into the objects so I used string. Also it was complaining about `double`and I don't have idea why.